src/Fmt_ast.ml

```diff
@@ -1521,7 +1521,9 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
                        (fmt_or_k
                           (Location.is_single_line pexp_loc c.conf.margin)
                           (fmt ")") (fits_breaks ")" "@ )"))
-                       (fits_breaks ")" "@,)") )
+                       (fmt_or_k
+                          (Location.is_single_line pexp_loc c.conf.margin)
+                          (fmt ")") (fits_breaks ")" "@,)")) )
```

Comment:
I'm not very happy about using `is_single_line` for this sort of thing, which is just laying out the parsetree, not trying to extract a bit more info from the concrete syntax to preserve some programmer intent. Keep in mind that every instance of `is_single_line` can lead to another whole round of formatting. There are already too many uses of `is_single_line` and related functions.